src/opm/utility/EModel.cpp

```diff
@@ -0,0 +1,474 @@
+/*
+   Copyright 2019 Equinor ASA.
+
+   This file is part of the Open Porous Media project (OPM).
+
+   OPM is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   OPM is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+   */
+
+#include <opm/utility/EModel.hpp>
+#include <opm/common/utility/FileSystem.hpp>
+
+#include <opm/common/utility/numeric/calculateCellVol.hpp>
+#include <string>
+#include <string.h>
+#include <sstream>
+#include <iterator>
+#include <iomanip>
+#include <algorithm>
+
+
+using EclEntry = std::tuple<std::string, Opm::EclIO::eclArrType, long int>;
+using ParamEntry = std::tuple<std::string, Opm::EclIO::eclArrType>;
+
+
+EModel::EModel(const std::string& filename)
+{
+    hasEgrid=false;
+
+    std::string rootN;
+
+    if (filename.substr(filename.size()-5,5)==".INIT") {
+        rootN=filename.substr(0,filename.size()-5);
+    } else {
+        rootN=filename;
+    }
+
+    if ( Opm::filesystem::exists(rootN + ".EGRID") ){
+        hasEgrid = true;
+        gridfile = std::make_unique<Opm::EclIO::EGrid>(rootN+".EGRID");
+    }
+
+    initfile = std::make_unique<Opm::EclIO::EclFile>(rootN+".INIT");
+
+    if (!initfile->hasKey("INTEHEAD")) {
+        std::string message="parameter INTEHEAD not found in selected init file " + rootN+".INIT";
+        throw std::invalid_argument(message);
+    }
+
+    std::vector<int> inteh=initfile->get<int>("INTEHEAD");
+
+    nI = inteh[8];
+    nJ = inteh[9];
+    nK = inteh[10];
+
+    nActive = static_cast<unsigned int>(inteh[11]);
```

Comment:
Use the proper grid!

Reply:
Minimum requirement for the EModel class is an INIT file. This class should be able work with a limited amount of functionality if an EGRID file is not existing.  Hence getting grid dimensions and and number of active cells from the INTEHEAD array in the init file is preferrd.

Reply:
I understand